### ISSUE TITLE
www-client/palemoon-32.4.0.1: newicon: command not found

### ISSUE THREAD

**Issue by sedimentation-fault** [deu/palemoon-overlay#139]
### Problem
Trying to build `www-client/palemoon-32.4.0.1`, all went smoothly, up to the install phase, specifically the `install_branding_files` function - where I got the errors
```
newicon: command not found
newmenu: command not found

```
from the environment  file at `/path-to-build-dir/portage/www-client/palemoon-32.4.0.1-r1/temp/environment`.

### Reason
This is an old system - last system update was in February 2022. This means my `portage` uses _eclasses_ from that time. Something has changed in _eclasses_ in-between: `eutils` does not support EAPI 6 or 7, for example, only 8. The ebuild must of course use the latest _eclasses_. It has:
```
EAPI=8
inherit palemoon-5 git-r3 flag-o-matic pax-utils
```
These, on an old system like mine, are not going to work anymore.

#### Upgrade the whole system
This is the `Gentoo` equivalent of the well-known "solution" from the `Windows` world: _reinstall everything, upgrade, reboot 100 times_ - yes, even **erase your disk and start from scratch**. Been there, will _not_ do that. Each time I upgrade, it costs me **2 months(!), 10 hours/day, 24x7**. Forget this "solution". I will upgrade when _I_ deem it appropriate, not when a browser forces me to.

#### Modify the ebuild and bring back the old EAPI and eutils
Replace the above with
```
EAPI=6
inherit palemoon-5 git-r3 flag-o-matic pax-utils eutils
```
`www-client/palemoon-32.4.0.1` now is built and installed without a problem. Works like a charm, as always - thank you.

### Epilog
Took me almost a day of experimentation - I thought I would write it somewhere for people with the same problem. It would be very desirable if the browser (or any piece of software, for that matter) did not dictate a whole system upgrade through a silent feature like the use of new _eclasses_.  I understand that `Gentoo` must go on and _ebuild_ maintainers must follow, but ideally this should happen painlessly, even for old systems.

You can mark this as solved. Thanks again for the good work.

**Comment by LanThrusteR** [comment]
I bumped into the same problem, newicon, newmenu are portage functions located in ./eclass/ folder in EAPI8 those functions were moved to desktop.eclass

Another solution to make it work is to open 
`./eclass/palemoon-5.eclass`

and add desktop to the first line 

`inherit check-reqs gnome2-utils toolchain-funcs xdg-utils`

so it would look like

`inherit check-reqs gnome2-utils toolchain-funcs xdg-utils desktop`

save palemoon-5.eclass and re-emege pale moon

**Comment by CaseOf** [comment]
are you ok to do a PR for this?
I'll merge it.

**Comment by l29ah** [comment]
Fixed in 93b270228bd2ac5b31f25e60cb1047726cab51b9

**Comment by l29ah** [comment]
In fact the author posted the bug several months after it was fixed, so i'm a bit puzzled.
